common/src/main/java/com/sequenceiq/cloudbreak/ccm/endpoint/BaseServiceEndpoint.java

```diff
@@ -0,0 +1,132 @@
+package com.sequenceiq.cloudbreak.ccm.endpoint;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Base class for service endpoint implementations.
+ */
+public class BaseServiceEndpoint implements ServiceEndpoint, Serializable {
+
+    /**
+     * The dummy port for URI construction.
+     */
+    private static final int DUMMY_PORT = -1;
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The host endpoint.
+     */
+    private final HostEndpoint hostEndpoint;
+
+    /**
+     * The optional port.
+     */
+    private final Integer port;
+
+    /**
+     * The optional URI.
```

Comment:
I'd like if these comments explained more. Example: URI for what?

Reply:
Improved comment